### PR TITLE
Apollo3: A collection of fixes and improvements

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

A collection of fixes and improvements for Apollo3 devices

### Testing Strategy

Using GPIO from a libtock-c application

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Review by hudson-ayers — COMMENTED** [review]
This does not pass CI currently.

Can you explain the directory renames in this PR?

**Comment by alistair23** [comment]
Ah, I included a debug statement. I'll remove that and fix the CI.

As for the renames, they are just moving some files that weren't correctly moved when the LoRa board was added. There are now two apollo3 boards and a bit of shared code between them. This just fixes up some paths

File: chips/apollo3/src/iom.rs [hunk 1]
```diff
@@ -628,52 +630,61 @@ impl<'a> Iom<'_> {
...
-                        if len % 4 == 1 {
-                            buf[len - 1] = write_buffer[self.write_index.get() + 0];
-                        } else if len % 4 == 2 {
-                            buf[len - 2] = write_buffer[self.write_index.get() + 0];
-                            buf[len - 1] = write_buffer[self.write_index.get() + 1];
-                        } else if len % 4 == 3 {
-                            buf[len - 3] = write_buffer[self.write_index.get() + 0];
-                            buf[len - 2] = write_buffer[self.write_index.get() + 1];
-                            buf[len - 1] = write_buffer[self.write_index.get() + 2];
+                        // Get an remaining data that isn't 4 bytes long
```
Comment (twilfredo) [inline chips/apollo3/src/iom.rs:668]:
```suggestion
                        // Get remaining data that isn't 4 bytes long
```

**Review by twilfredo — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**